ts-client/src/dlmm/index.ts

```diff
@@ -3623,6 +3629,339 @@ export class DLMM {
     );
   }
 
+  /**
+   * The `seedLiquidity` function create multiple transactions, which each transaction consists of initialize position, and add liquidity instruction. The combination of liquidity deposited to the position will be the ilm / lfg curve
+   * @param
+   *
+   */
+  public async seedLiquidity(
+    owner: PublicKey,
+    operator: PublicKey,
+    feeOwner: PublicKey,
+    seedAmount: BN,
+    curvature: number,
+    minPrice: number,
+    maxPrice: number,
+    base: PublicKey
+  ): Promise<TransactionInstruction[][]> {
+    const toLamportMultiplier = new Decimal(
+      10 ** (this.tokenY.decimal - this.tokenX.decimal)
+    );
+
+    const minPricePerLamport = new Decimal(minPrice).mul(toLamportMultiplier);
+    const maxPricePerLamport = new Decimal(maxPrice).mul(toLamportMultiplier);
+
+    const minBinId = new BN(
+      DLMM.getBinIdFromPrice(minPricePerLamport, this.lbPair.binStep, false)
+    );
+
+    const maxBinId = new BN(
+      DLMM.getBinIdFromPrice(maxPricePerLamport, this.lbPair.binStep, true)
+    );
+
+    if (minBinId.toNumber() < this.lbPair.activeId) {
+      throw new Error("minPrice < current pair price");
+    }
+
+    if (minBinId.toNumber() >= maxBinId.toNumber()) {
+      throw new Error("Price range too small");
+    }
+
+    // Generate amount for each bin
+    const k = 1.0 / curvature;
+    const binDepositAmount = new Map<number, BN>();
+
+    let minAmount = new BN(new Decimal(2).pow(64).sub(1).toString());
+    let maxAmount = new BN(0);
+
+    for (let i = minBinId.toNumber(); i < maxBinId.toNumber(); i++) {
+      const c1 = getC(
+        seedAmount,
+        this.lbPair.binStep,
+        new BN(i + 1),
+        this.tokenX.decimal,
+        this.tokenY.decimal,
+        minPrice,
+        maxPrice,
+        k
+      );
+
+      const c0 = getC(
+        seedAmount,
+        this.lbPair.binStep,
+        new BN(i),
+        this.tokenX.decimal,
+        this.tokenY.decimal,
+        minPrice,
+        maxPrice,
+        k
+      );
+
+      const amount = new BN(c1.sub(c0).floor().toString());
+
+      if (amount.lt(minAmount)) {
+        minAmount = amount;
+      }
+
+      if (amount.gt(maxAmount)) {
+        maxAmount = amount;
+      }
+
+      binDepositAmount.set(i, amount);
+    }
+
+    // Look for multiplier with least loss
+    let decompressMultiplier = new BN(10);
```

Comment:
I thought we just go with min as 10^token_decimals?

Reply:
Fixed, seems like this is more simplier.